lib/elixir/lib/gen_server.ex

```diff
@@ -566,6 +566,10 @@ defmodule GenServer do
   """
   @spec call(server, term, timeout) :: term
   def call(server, request, timeout \\ 5000) do
+    if server == self do
+      raise ArgumentError, message: 
+      "invalid server argument in call(server, request, timeout). Looks like you trying to call a parent process"
```

Comment:
Either we should use an atom and handle it along with `:normal`, `:timeout` and other atom reasons in `Exception` or we should use the reason `{%ArgumentError{}, stacktrace}` and the error message/stacktrace is included in the message.